### PR TITLE
Update support policy, tooling, and docs

### PR DESCRIPTION
**Pull request by danyalaytekin**
This PR goes some way to update the tooling used in this project:

1. Replace Grunt tasks with npm scripts
2. Replace JSHint with ESLint and [pa11y-lint-config](https://github.com/pa11y/pa11y-lint-config)
3. FIx lockfile version to `2`
4. Publish with minimum supported version of Node (18)

It also:

1. Documents how to use `nektos/act` to validate/test the workflows.  This is a simpler project than most of the others so `act` works fine here.
2. Adds our [chosen support policy](https://github.com/pa11y/org/issues/18).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -136,40 +136,64 @@ If you'd like to contribute code, get started by cloning the repo and running `n
...
+| :-------------- | :------------------ | :----------------------- | :--------------- |
+| `3`             |                     | `>= 12`                  | ✅ Current major version |
+| `2`             | `2.0`               | `8`, `10`                | 2022-05-26       |
+| `1`             | `1.2`               | `0.10`, `0.12`, `4`, `6` | 2020-01-05       |
 
 ## License
 
-Licensed under the [GNU General Public License 3.0](LICENSE.txt).<br/>
-Copyright &copy; 2013–2023, Team Pa11y
+Licensed under the [GNU General Public License 3.0][info-license].
```
Comment (danyalaytekin) [inline README.md:193]: Updated to use the link definition (which was pointing to the same file)... but I noticed that:

- the badge and its text `say LGPL` (lines 8 and 204 of this file)
- this line, `LICENSE`, and all the code file headers say `GPL`

Which should it be?

Comment (hollsk) [reply README.md:193]: mm yes, confusing. I've searched the Slack archives and the Slack archives tell me that LGPL is correct, even though it's the least-used in the codebase 😅

**Review by hollsk — APPROVED** [review]
Alright this looks good to me. I note your question about LGPL vs GPL, and to the best of my knowledge, LGPL is the right one. I'm just going to approve this PR, and we can just merge and then rationalise the licensing later (might need to go through more repos to see if there are more inconsistencies, so makes sense to me to do it in a batch)

Thanks!!

File: README.md (hunk 1 above)
Comment (danyalaytekin) [reply README.md:193]: Ok thanks @hollsk, I'll make a little PR after this one to update it.

**Pull request merged**